### DISCUSSION TITLE
Problems when trying to display the values from a dataset

### DISCUSSION THREAD

**Discussion by jgalan** [rest-for-physics/framework#488] [Q&A] [answered]
I am having difficulties while printing the column data of the data frame defined inside the dataset.

```
The dataset is ready. You may now access the dataset using:

 - dSet->PrintMetadata()
 - dSet->GetDataFrame().GetColumnNames()
 - dSet->GetTree()->GetEntries()

root [0] ROOT::DisableImplicitMT()
root [1] dSet->GetDataFrame().Display({"final_posX"})->Print()
Error in <TRint::HandleTermInput()>: std::runtime_error caught: RLoopManager::Run: when the RDataFrame was constructed the number of slots required was 4, but when starting the event loop it was 1. Maybe DisableImplicitMT() was called after the RDataFrame was constructed?
```

Maybe @juanangp you got some insights on how to solve this?

**Comment by juanangp** [comment, marked as answer]
I think the issue is that the dataSet has been built with multiThreading enabled, looking at some ROOT forums there is not much to do while using `Display` with MT enabled.

I suggest the following:
- Make configurable to enable or disable MT at `TRestDataSet::GenerateDataSet` (default be enabled by default)
- Make configurable enable or disable MT at `Import` e.g. `Import(const std::string& fileName, bool enableMT = true)`

You should be able to do `Display` with a dataset which has been generated using MT enabled but it is imported with MT disabled.
